File: dev/extension-examples/echo_agent.py
#!/usr/bin/env python3
from loop_agent import LoopAgent


class EchoAgent(LoopAgent):
    name = "echo-agent"
    version = "0.1.0"

    def run(self, message: str, run_id: str):
        yield f"Echo: {message}\n"


if __name__ == "__main__":
    EchoAgent().serve()

File: dev/extension-examples/ts/echo_agent.ts
import { LoopAgent } from './loop_agent'

class EchoAgent extends LoopAgent {
  name = 'echo-agent'
  version = '0.1.0'

  async *run(message: string, _runId: string) {
    yield `Echo: ${message}\n`
  }
}

new EchoAgent().serve()
